src/sidebar/store/modules/frames.ts

```diff
@@ -147,7 +150,20 @@ const mainFrame = createSelector(
 
   // Sub-frames will all have a "frame identifier" set. The main frame is the
   // one with a `null` id.
-  frames => frames.find(f => !f.id) || null,
+  firstFrameWithoutId,
+);
+
+/**
+ * Return the "main" frame that the sidebar is connected to, falling back to the
+ * first available frame.
+ * For VitalSource books, we don't have a mainFrame, but we do have a single
+ * frame on the list.
```

Comment:
This describes the implementation, but I'd like to try and reword this to differentiate it a bit more clearly from `mainFrame`. How about:

"Return the default frame that annotations are assumed to be associated with, if they can't be matched to a more specific frame.

For most document types this is the same as ({@see mainFrame}), but for eg. VitalSource books there is no main frame, but instead only a frame with the current chapter content." 

This aligns with the similar logic in `frameForAnnotation` in `services/frame-sync.ts`.